File: practice-01/exercise-01/go/main.go
package main

import (
	"fmt"
	"math/rand"
)

const N = 20

func load(a *[N]int) {
	for i := range N {
		a[i] = rand.Intn(100)
	}
}

func show(a *[N]int) {
	fmt.Println(*a)
}

func checkZeros(a *[N]int) {
	var i = 0
	for i < N && a[i] != 0 {
		i++
	}
	if i < N {
		fmt.Println("Hay al menos un cero en el arreglo.")
	} else {
		fmt.Println("No hay ceros en el arreglo.")
	}
}

func printOddPositions(a *[N]int) {
	fmt.Print("Odd Position values: ")
	for i := range N {
		if i%2 == 0 {
			fmt.Printf("%d ", a[i])
		}
	}
}

func countOddValues(a *[N]int) {
	var count = 0

	for i := range N {
		if a[i]%2 == 0 {
			count++
		}
	}
	fmt.Printf("\nCount of odd nums: %d\n", count)
}

func main() {
	var arr [N]int
	load(&arr)
	show(&arr)
	checkZeros(&arr)
	printOddPositions(&arr)
	countOddValues(&arr)
}

File: practice-01/exercise-01/c/main.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#define N 20

void load(int a[])
{
  srand(time(NULL));
  for (int i = 0; i < N; i++)
  {
    a[i] = rand() % 100;
  }
}

void showEvenPosition(int a[])
{
  for (int i = 0; i < N; i++)
    if (i % 2 == 0)
      printf("%d, ", a[i]);
}

void countEvens(int a[])
{
  int count = 0;
  for (int i = 0; i < N; i++)
    if (a[i] % 2 == 0)
      count++;

  printf("Cantidad de números pares: %d", count);
}

void show(int a[])
{
  for (int i = 0; i < N; i++)
    printf("%d, ", a[i]);
}

int checkZeros(int a[])
{
  int i = 0;
  while (i < N && a[i] != 0)
    i++;
  if (i < N)
  {
    printf("Hay al menos un cero en el array.");
  }
  else
  {
    printf("No hay ceros en el array.");
  }
}

int main()
{
  int arr[N];
  load(arr);
  printf("\n");
  show(arr);
  printf("\n");
  checkZeros(arr);
  printf("\n");
  showEvenPosition(arr);
  printf("\n");
  countEvens(arr);
  printf("\n");
  return 0;
}